go/pkg/cosmos/history.go

```diff
@@ -49,6 +49,27 @@ func NewDefaultSources(client *HTTPClient, pubkey string, formatTx func(*coretyp
 	}
 }
 
+func ValidatorHistorySources(client *HTTPClient, pubkey string, formatTx func(*coretypes.ResultTx) (*Tx, error)) map[string]*TxState {
+	request := func(query string, page int, pageSize int) ([]HistoryTx, error) {
+		result, err := client.TxSearch(query, page, pageSize)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+
+		txs := []HistoryTx{}
+		for _, tx := range result.Txs {
+			txs = append(txs, &ResultTx{ResultTx: tx, formatTx: formatTx})
+		}
+
+		return txs, nil
+	}
+
+	return map[string]*TxState{
+		"send":    NewTxState(true, fmt.Sprintf(`"delegate.validator='%s'"`, pubkey), request),
+		"receive": NewTxState(true, fmt.Sprintf(`"unbond.validator='%s'"`, pubkey), request),
+	}
+}
+
```

Comment:
Ideally the messages to parse would be passed in from the outside but I don't want to mess with data flow too much at this point

Reply:
@kaladinlight Btw how did you establish the messages to use here? I've checked all of Tendermind RPC api docs as well as Figment ones but I didn't find any detailed overview of the data model used in responses

Reply:
When you say messages, are you referring to the query string param associated with the tx event (ie. `delegate.validator`)?

Reply:
These events are not really documented to my knowledge. I either just look at an example transaction and inspect the `logs[].events[]` payload for a specific value needed as the search query:

![image](https://user-images.githubusercontent.com/35275952/200635788-d2c78db3-349b-4361-9e14-e37d56fb5008.png)

You can see type `delegate` and key `validator`. All events are indexed by default using `type.key` as lookup (ie. `delegate.validator`).

Alternatively, you can view the source code directly. Each module will declare events: https://github.com/cosmos/cosmos-sdk/blob/ff4f995b185f0b7e318bd0a47d0b1aa824406d8b/x/staking/types/events.go#L9

and you can then look for where the actual event is being created with the associated key/val pairs: https://github.com/cosmos/cosmos-sdk/blob/ff4f995b185f0b7e318bd0a47d0b1aa824406d8b/x/staking/keeper/msg_server.go#L242